src/main/java/com/patreon/PatreonAPI.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param accessToken The "Creator's Access Token" found on <a href="https://www.patreon.com/platform/documentation/clients">the patreon client page</a> <b>OR</b> OAuth access token
+     */
+    public PatreonAPI(String accessToken) {
+        this.accessToken = accessToken;
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
+        this.converter = new ResourceConverter(
+            objectMapper,
+            User.class,
+            Campaign.class,
+            Pledge.class
+        );
+        this.converter.enableDeserializationOption(DeserializationFeature.ALLOW_UNKNOWN_INCLUSIONS);
+    }
+
+    /**
+     * Get the user object of the creator
+     *
+     * @return JSONAPIDocument<User> containing all data pertaining to the current user
+     * @throws IOException Thrown when the GET request failed
+     */
+    public JSONAPIDocument<User> fetchUser() throws IOException {
+        return converter.readDocument(
+            getDataStream("current_user"),
+            User.class
+        );
+    }
+
+    /**
+     * Get a list of campaigns the current creator is running - also contains other related data like Goals
+     * Note: The first campaign data object is located at index 0 in the data list
+     *
+     * @return JSONAPIDocument<List<Campaign>> containing the above-mentioned data
+     * @throws IOException Thrown when the GET request failed
+     */
+    public JSONAPIDocument<List<Campaign>> fetchCampaigns() throws IOException {
+        String path = new URIBuilder()
+            .setPath("current_user/campaigns")
+            .addParameter("include", "rewards,creator,goals")
+            .toString();
+        return converter.readDocumentCollection(
+            getDataStream(path),
+            Campaign.class
+        );
+    }
+
+    /**
+     * Retrieve pledges for the specified campaign
+     *
+     * @param campaignId id for campaign to retrieve
+     * @param pageSize   how many pledges to return
+     * @param pageCursor ignore, put null.
+     * @return JSONAPIDocument<List<Pledge>> containing pledges & associated data
+     * @throws IOException Thrown when the GET request failed
+     */
+    public JSONAPIDocument<List<Pledge>> fetchPageOfPledges(String campaignId, int pageSize, String pageCursor) throws IOException {
+        URIBuilder pathBuilder = new URIBuilder()
+            .setPath(String.format("campaigns/%s/pledges", campaignId))
+            .addParameter("page[count]", String.valueOf(pageSize));
+        if (pageCursor != null) {
+            pathBuilder.addParameter("page[cursor]", pageCursor);
+        }
+        return converter.readDocumentCollection(
+            getDataStream(pathBuilder.toString()),
+            Pledge.class
+        );
+    }
+
+    public String getCursorFromPledgesResponse(JSONAPIDocument<List<Pledge>> pledgesResponse) {
```

Comment:
🥇

Reply:
optional, possibly in a later PR: it would be really cool to make this resource-independent.